src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -198,8 +225,10 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
     this.setLocally(key, newVal);
 
     try {
-      const { settings } = (await this.api.batchSet(key, newVal)) || {};
-      this.cache = defaultsDeep({}, defaults, settings);
+      const { settings } = this.cache[key]?.preferBrowserSetting
+        ? this.setBrowserStoredSettings(key, newVal)
+        : (await this.api.batchSet(key, newVal)) || {};
+      this.cache = defaultsDeep({}, defaults, this.getBrowserStoredSettings(), settings);
```

Comment:
That's the correct interpretation, and the intended behavior.

> default is `darkMode: true`

Yep, and this is coming from the configs - under this new implementation, there's no longer the UI ability to set a cluster-wide default.

> I have `preferBrowserSetting: false`

This setting is irrelevant for the purpose of this client and not read. It's only purpose is to change/update the localStorage settings on bootstrap/startup

> my browser's localStorage also says `darkMode: false`

This is the source of truth for any UISetting that has the `preferBrowserSetting` flag prop set to `true`. As an implementation detail, the bootstrap/startup script are responsible for making sure these values are set (and setting them correctly based on defaults or preferred browser settings.

> My savedObjects doesn't have a `DarkMode`.

Yep - in the current implementation, we wouldn't look there. We may change this in the future so that we use/store a persistent source of truth there which can be utilized by the bootstrap/startup initialization; but the source of truth would remain the local storage value.

Reply:
> That's the correct interpretation, and the intended behavior.

@joshuarrrr, when `preferBrowserSetting: false`, I would expect nothing from the browser to be honored at all; i would expect that the default `darkMode: true` will be applied.

Reply:
is this a blocker @AMoo-Miki ?

Reply:
Rocky, I think the unexpected result will confuse users enough to make it a problem.